sdk/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -626,8 +626,24 @@ public override async Task RemoveAllConfigAsync(string stackName, IEnumerable<st
         /// <inheritdoc/>
         public override async Task<WhoAmIResult> WhoAmIAsync(CancellationToken cancellationToken = default)
         {
-            var result = await this.RunCommandAsync(new[] { "whoami" }, cancellationToken).ConfigureAwait(false);
-            return new WhoAmIResult(result.StandardOutput.Trim());
+            var version = await this.RunCommandAsync(new[] { "version" }, cancellationToken).ConfigureAwait(false);
+            if (!SemVersion.TryParse(version.StandardOutput.Trim(), SemVersionStyles.AllowLowerV, out var semVersion))
+            {
+                throw new Exception("Could not parse pulumi CLI version");
+            }
```

Comment:
I think if the version output fails to parse, we should just assume "old version".

Reply:
Reasonable